documentation/docs/configuration.md

```diff
@@ -62,6 +62,99 @@ steps:
+  | info | INFO | NOTICE |
+  | debug | INFO | NOTICE |
+  | warn | WARNING | ALERT |
+  | error | ERROR | EXCEPTION |
+  | fatal | FATAL | EXCEPTION |
+  | panic | FATAL | EXCEPTION |
+
+- `subject`: short description of the event (defaults to the step name, but
+  can be overwritten with the event template)
+- `body`: the log message
+- `priority`: (optional) an integer number in the range [1:1000] (not set by
+  "Piper", but can be set with the event template)
+- `tags`: optional key-value pairs. The following are set by "Piper":
+  - `logLevel`: the "Piper" log level
+  - `ans:correlationId`: a unique correlation ID of the pipeline run
+    (defaults to the URL of that pipeline run, but can be overwritten with the
+    event template)
+  - `ans:sourceEventId`: also set to the "Piper" correlation ID (can also be
+    overwritten with the event template)
+- `resource`: the following default properties are set by "Piper":
+  - `resourceType`: resource type identifier (defaults to 'Pipeline', but
+    can be overwritten with the event template)
+  - `resourceName`: unique resource name (defaults to 'Pipeline', can be
+    overwritten with the event template)
+  - `resourceInstance`: (optional) resource instance identifier (not set by
+    "Piper", can be set with the event template)
+
+The following event properties cannot be set and are instead set by the SAP
+Alert Notification service:
+`region`, `regionType`, `resource.globalAccount`, `resource.subAccount` and
+`resource.resourceGroup`
+
+For more information and an example of the structure of an alert
+notification service event, see
+[SAP Alert Notification Service Events](https://help.sap.com/viewer/5967a369d4b74f7a9c2b91f5df8e6ab6/Cloud/en-US/eaaa37e6ff62486ebb849507dc33abc6.html)
+on SAP Help Portal.
+
+### SAP Alert Notification service configuration
+
+The SAP Alert Notification service service-key needs to be present in the
+environment, where the "Piper" binary is run. The environment variable used
+is: `PIPER_ansServiceKey`
+
+If Jenkins is used to run "Piper", you can use the Jenkins credential store
+to store the alert notification service service-key as a "Secret Text"
+credential. Provide the credential id in the configuration file as follows:
+
+```yaml
+hooks:
+  ans:
+    serviceKeyCredentialsId: 'my_ANS_Service_Key'
+```
+
+You can also create an event template in JSON format to overwrite or add
+certain event details. There are two ways to do this:
+
+1. Provide an event template file in JSON format and set the
+   `eventTemplateFilePath` parameter in the config file as follows:
+
+   ```yaml
+   hooks:
+     ans:
+       eventTemplateFilePath: './path/to/my/eventTemplate.json'
+   ```
+
+2. Provide the JSON string directly in the environment where the "Piper"
+   binary is run. The environment variable used in this case is:
+   `PIPER_ansEventTemplate`. For example in unix:
+
+    ```bash
+    export PIPER_ansEventTemplate='{"priority": 999}'
+    ```
+
```

Comment:
Add explanation of the default event template, being overwritten by the event template from file, being in turn overwritten be the event template from the environment.

Reply:
done